example/spec/parallel/lzardsBackup/LzardsBackupSpec.js

```diff
@@ -172,27 +176,61 @@ describe('The Lzards Backup Task ', () => {
       beforeAllFailed = true;
       throw error;
     }
-  });
+  };
 
-  it('succeeds', () => {
-    if (beforeAllFailed) fail('beforeAll() failed');
-    else {
-      expect(lzardsBackupOutput.FunctionError).toBe(undefined);
-    }
-  });
+  describe('The Lzards Backup Task with override configured', () => {
+    it('invokes successfully', async () => {
+      const taskConfig = {
+        lzardsProvider: 'BOGUS_PROVIDER',
+        cumulus_message: {
+          outputs: [
+            {
+              source: '{$.originalPayload}',
+              destination: '{$.payload}',
+            },
+            {
+              source: '{$.backupResults}',
+              destination: '{$.meta.backupStatus}',
+            },
+          ],
+        },
+      };
 
-  it('has the expected backup information', () => {
-    const backupStatus = JSON.parse(new TextDecoder('utf-8').decode(lzardsBackupOutput.Payload)).meta.backupStatus;
-    console.log(`backupStatus: ${JSON.stringify(backupStatus)}`);
-    expect(backupStatus[0].status).toBe('COMPLETED');
-    expect(backupStatus[0].statusCode).toBe(201);
-    expect(backupStatus[0].granuleId).toBe(granuleId);
-    expect(backupStatus[0].provider).toBe(provider);
-    expect(backupStatus[0].createdAt).toBe(tenMinutesAgo);
-    expect(backupStatus[0].collectionId).toBe(encodedConstructCollectionId(collection.name, collection.version));
+      await testSetup({ task_config: taskConfig });
+      if (beforeAllFailed) fail('beforeAll() failed');
+      else {
+        expect(lzardsBackupOutput.FunctionError).toBe(undefined);
+      }
+    });
+
+    it('has the expected backup information', () => {
+      const backupStatus = JSON.parse(new TextDecoder('utf-8').decode(lzardsBackupOutput.Payload)).meta.backupStatus;
+      console.log(`backupStatus: ${JSON.stringify(backupStatus)}`);
+      expect(backupStatus[0].status).toBe('FAILED');
+      expect(backupStatus[0].body).toContain('Unprocessable Entity');
+    });
   });
 
-  describe('The Lzards API Client', () => {
+  describe('The Lzards Backup Task', () => {
+    it('invokes successfully', async () => {
+      await testSetup();
+      if (beforeAllFailed) fail('beforeAll() failed');
+      else {
+        expect(lzardsBackupOutput.FunctionError).toBe(undefined);
+      }
+    });
+
+    it('has the expected backup information', () => {
+      const backupStatus = JSON.parse(new TextDecoder('utf-8').decode(lzardsBackupOutput.Payload)).meta.backupStatus;
```

Comment:
I think we need to check `if (beforeAllFailed) fail('beforeAll() failed');` in each assertion?

Reply:
Good catch.   What I wouldn't give for function decorators in node. 

5890438440